services/pull/pull.go

```diff
@@ -372,19 +372,29 @@ func checkForInvalidation(ctx context.Context, requests issues_model.PullRequest
 	return nil
 }
 
+type TestPullRequestOptions struct {
+	RepoID      int64
+	Doer        *user_model.User
+	Branch      string
+	IsSync      bool
```

Comment:
What does `IsSycn` mean? Could it have some comments?

Reply:
`IsSync` means if it's a pull request update.

Reply:
Then what does `IsSync=false` mean? Which action triggers `IsSync=false`? Could it have some comments?

Reply:
I have added comments in f8b02f7. Looks like `AddTestPullRequestTask` needs a big refactor. So that I don't want to introduce it to this PR.

Reply:
By that comment, I can understand if `IsSync=true` then `it's a pull request synchronization`.

But if `IsSync=false`, then what it is?Which action triggers `IsSync=false`?

Reply:
4961418dda
This maybe should be an action parameter. It could be pull request synchronization, updating or merging.